specs/v2-sketch.yaml

```diff
@@ -0,0 +1,365 @@
+          example: ["username@example.com"]
+          items:
+            type: string
+        selected_email:
+          type: string
+          example: "username@example.com"
+        avatar_url:
+          type: string
+          example: "https://avatars.example.com/369ea08d7172bfc72207e2ae1c1b17ef"
+# github_id
+# bitbucket
+# bitbucket-authorized
+# pusher-id
+#   These all feel far too specific, should be rolled into "identities", right?
+#
+# github-oauth-scopes
+#   don't wanna include GitHub oauth scopes
+#
+# heroku-api-key
+#   Nope, not including credentials, obfuscated or not
+        identities:
+          type: object
+          additionalProperties:
+            type: object
+            properties:
+              avatar_url:
+                type: string
+                example: "https://avatars.example.com/369ea08d7172bfc72207e2ae1c1b17ef"
+              external_id:
+                anyOf:
+                  - type: string
+                  - type: number
+                example: 1234567
+              id:
+                anyOf:
+                  - type: string
+                  - type: number
+                example: 1234567
+              name:
+                type: string
+                example: "User Name"
+              user?:
+                type: boolean
+              domain:
+                type: string
+                example: "github.com"
+              type:
+                type: string
+                example: "github"
+              authorized?:
+                type: boolean
+              provider_id:
+                type: string
+                format: uuid
+              login:
+                type: string
+                example: "username"
+    Actor:
+      type: object
+      properties:
+        login:
+          type: string
+        name:
+          type: string
+        id:
+          type: string
+          format: uuid
+    Project:
+      type: object
+      properties:
+        id:
+          type: string
+          format: uuid
+        # add a link to the project, Pat suggested we consider link headers
+    # Elide from Workflow for the time being. If we can get pipelines in weeks we don't need this. Backup plan if people shout is to put this into the Workflow response.
+    TriggerSummary:
+      type: object
+      properties:
+        type:
+          type: string
+          enum: ["webhook", "api", "schedule"]
+        actor:
+          $ref: "#/components/schemas/Actor"
+        vcs:
+          type: object
+          properties:
+            branch:
+              type: object
+              properties:
+                name:
+                  type: string
+                default_branch:
+                  type: boolean
+            tag:
+              type: string
+            commit:
+              type: string
+    Workflow:
+      type: object
+      required:
+        - id
+        - name
+        - created_at
+        - project
+        - errors
```

Comment:
I don't see this field represented in properties, did I miss it?

Reply:
I forgot to update the required fields list when `errors` was removed.

Errors on a workflow only happen due to bad config. Since the config is already validated upstream for workflows triggered via pipelines it is an exceedingly unlikely event so I think we're better off handling it by not creating workflows and reporting that upstream.